### PR TITLE
update smart contract

### PR DESCRIPTION
**Pull request by webelf101**

### CODE REVIEW

**Review by taariq — APPROVED** [review]
Looks like a great. start. @measure-fi you okay to give a check?

File: contracts/limit_order_uniswap_v2.vy [hunk 1]
```diff
@@ -97,102 +78,84 @@ def _safe_transfer_from(_token: address, _from: address, _to: address, _value: u
...
-        self._safe_approve(WETH, ROUTER, deposit.amount0)
-    else:
-        self._safe_approve(deposit.token0, ROUTER, deposit.amount0)
-    amounts: DynArray[uint256, 2] = [0, 0]
-    if deposit.token1 == VETH:
-        if profit_taking_or_stop_loss:
-            amounts = UniswapV2Router(ROUTER).swapExactTokensForETH(deposit.amount0, deposit.amount1_max, [deposit.token0, WETH], deposit.depositor, block.timestamp)
-        else:
-            amounts = UniswapV2Router(ROUTER).swapExactTokensForETH(deposit.amount0, deposit.amount1_min, [deposit.token0, WETH], deposit.depositor, block.timestamp)
+    assert deposit.amount1 > 0
```
Comment (measure-fi) [inline contracts/limit_order_uniswap_v2.vy:120]: This assert is above in the diff as `assert deposit.amount0 > 0`. I'm not actually sure what `amount0` and `amount 1` represent semantically though.

Comment (webelf101) [reply contracts/limit_order_uniswap_v2.vy:120]: Because the swap path is reverted.

File: README.md [hunk 2]
```diff
@@ -53,39 +53,42 @@ brownie run scripts/deploy_pancakeswap.py --network bsc-main
...
-| token1      | address | Expected token address                  |
-| amount0     | uint256 | Deposit token amount                    |
-| amount1_min | uint256 | Expected token amount for stop loss     |
-| amount1_max | uint256 | Expected token amount for profit taking |
+| Key           | Type      | Description                                 |
+| ------------- | --------- | ------------------------------------------- |
+| path          | address[] | Initial token swap path via Uniswap V2      |
+| amount0       | uint256   | Deposit token amount                        |
+| min_amount1   | uint256   | Expected token amount from the initial swap |
+| profit_taking | uint256   | Permille of profit_taking                   |
```
Comment (verabehr) [inline README.md:61]: why is it permille and not percent?

Comment (webelf101) [reply README.md:61]: I want to set profit taking as 5.5% maybe?

**Pull request merged**